packages/eslint-plugin/lib/config/package-json.ts

```diff
@@ -1,34 +1,31 @@
 import type { Linter } from "eslint";
+import { concat } from "eslint-flat-config-utils";
+import packageJsonPluginRecommended from "eslint-plugin-package-json/configs/recommended";
 
-const config: Linter.Config = {
-    overrides: [
-        {
-            files: ["package.json"],
-            plugins: ["package-json"],
-            extends: ["plugin:package-json/recommended"],
-            parser: "jsonc-eslint-parser",
-            rules: {
-                "package-json/prefer-repository-shorthand": "off",
-                "package-json/sort-collections": [
-                    "error",
-                    [
-                        // Do not sort "scripts".
-                        "devDependencies",
-                        "dependencies",
-                        "peerDependencies",
-                        "config"
-                    ]
-                ],
-                // Doesn't support "workspace:*" at the moment.
-                "package-json/valid-package-def": "off",
-                // I am not sure why, this rule is triggering errors for valid paths.
-                "package-json/valid-repository-directory": "off"
-            }
+const config: Linter.FlatConfig[] = await concat(
+    {
+        ...(packageJsonPluginRecommended as Linter.FlatConfig),
+        name: "Workleap/Package.json",
+        files: ["package.json"],
+        rules: {
+            ...packageJsonPluginRecommended.rules,
```

Comment:
Isn't the "concat" already taking care of merging the rules?

Reply:
No, concat just merges and flattens arrays, it doesn't do a deep merge. It still produces an array of config objects.